internal/manager/indices.go

```diff
@@ -0,0 +1,66 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	IdxPodsByComposition           = ".podsByComposition"
```

Comment:
It's not important to include in this PR, but I'm thinking it would be nice to make these private and export funcs that return the client.MatchingFields for a given index while taking a structured input like a composition, etc. Kind of keeps the dynamic typing of the indices confined to this package.